caikit/runtime/servicers/info_servicer.py

```diff
@@ -38,12 +44,65 @@ class InfoServicer:
     """This class contains the implementation for retrieving information about the
     library and services."""
 
+    def GetModelsInfo(
+        self, request: ModelInfoRequest, context  # pylint: disable=unused-argument
+    ) -> ModelInfoResponse:
+        """Get information on the loaded models for the GRPC server
+
+        Args:
+            request: ModelInfoRequest
+            context
+
+        Returns:
+            models_info: ModelInfoResponse
+                DataObject containing the model info
+        """
+        return self.get_models_info_impl().to_proto()
+
+    def get_models_info(self) -> Dict[str, List[Dict[str, Union[str, int]]]]:
+        """Get information on models for the HTTP server
+
+        Returns:
+            model_info_dict: Dict[str, List[Dict[str, str]]]
+                Dict representation of ModelInfoResponse
+        """
+        return self.get_models_info_impl().to_dict()
+
+    def get_models_info_impl(self) -> ModelInfoResponse:
+        """Helper function to get the list of models
+
+        Returns:
+            model_info: ModelInfoResponse
+                DataObject with model information
+        """
+        model_manager = ModelManager.get_instance()
+
+        # Get all loaded models
+        response = ModelInfoResponse(models=[])
+        for name, loaded_module in model_manager.loaded_models.items():
+            model_instance = loaded_module.model()
+            response.models.append(
```

Comment:
Here's my proposal to make this more flexible going forward (could be in a follow-on PR):

1. Add a `def get_metadata(self) -> Dict[str, str]` method to `ModuleBase`
2. Add a `metadata: Dict[str, str]` field to `ModelInfo` that defaults to `{}`
3. Populate (2) from (1) here if (2) returns anything

This would allow for custom logic in things like remote TGIS implementations of `text_generation`.

Reply:
@HonakerM @gabe-l-hart the metadata property already exists?
```
    @property
    def metadata(self) -> Dict[str, Any]:
```

Reply:
@joerunde I saw that but I think having a separate `get_metadata` function would be useful. The `metadata` property returns the full Moduleconfig that was used during loading and I don't think we want to return that (especially if weights/info could be proprietary)

Reply:
Ok, interesting. I think we should change the name of `get_metadata` to `model_info` then to avoid the naming confusion